js/components/portal-dashboard/answers/iframe-answer.tsx

```diff
@@ -0,0 +1,136 @@
+    const baseUrl = `${window.location.origin}${window.location.pathname}`;
+    const params = queryString.parse(window.location.search);
+    params.studentId = answer.get("platformUserId");
+    params.iframeQuestionId = question.get("id");
+    const newSearch = queryString.stringify(params);
+    return `${baseUrl}?${newSearch}`;
+  }
+
+  renderLink() {
+    const { answer, question } = this.props;
+    const { iframeVisible } = this.state;
+    const linkUrl = this.getLinkURL(answer.get("answer"));
+    const externalLinkIcon = <span className="pr-icon-external-link" />;
+    if (question?.get("displayInIframe")) {
+      const toggleText = iframeVisible ? "Hide" : "View Work";
+      const standaloneLinkUrl = this.getStandaloneLinkUrl(question, answer);
+      return (
+        <React.Fragment>
+          <a onClick={this.toggleIframe} target="_blank" data-cy="toggleIframe">{toggleText}</a> |{" "}
+          <a href={standaloneLinkUrl} target="_blank" data-cy="standaloneIframe">Open in new tab {externalLinkIcon}</a>
+        </React.Fragment>
+      );
+    } else {
+      return <a href={linkUrl} target="_blank" data-cy="externalIframe">View work in new tab {externalLinkIcon}</a>;
+    }
+  }
+
+  renderIframe() {
+    const { answer, question, responsive } = this.props;
+    let url;
+    let state;
+    // There are two supported answer types handled by iframe question: simple link or interactive state.
+    if (answer.get("type") === "external_link") {
+      // Answer field is just the reportable URL. We don't need any state.
+      url = answer.get("answer");
+      state = null;
+    } else if (answer.get("type") === "interactive_state") {
+      // URL field is provided by question. Answer field is a state that will be passed
+      // to the iframe using iframe-phone.
+      url = question?.get("url");
+      state = answer.get("answer");
+    }
+    return (
+      <div className={`iframe-answer-content ${responsive ? "responsive" : ""}`}>
+        <InteractiveIframe src={url} state={state} width={question?.get("width")} height={question?.get("height")} />
+      </div>
+    );
+  }
+
+  shouldRenderIframe() {
+    const { question } = this.props;
+    const { iframeVisible } = this.state;
+
+    if (question?.get("displayInIframe")) {
+      return iframeVisible;
+    } else {
+      return false;
+    }
+  }
+
+  render() {
+    const { showFullAnswer, answer, responsive } = this.props;
+    const answerText = answer.get("answerText");
+
+    return (
+      <div className={`iframe-answer ${responsive ? "responsive" : ""}`} data-cy="iframe-answer">
+        <div className={`iframe-answer-header ${responsive ? "responsive" : ""}`}>
+          { answerText && <div>{ answerText }</div> }
+          {/* This assumes only scaffolded questions and fill in the blank questions have answerTexts */}
+          { (showFullAnswer && !answerText)? this.renderLink() : ""}
```

Comment:
I don't think we want to make an entire copy of 'iframe-answer.js` just to change this one line of code.  When we first investigated this a month or so ago, I was under the impression that we might need to rewrite some of these answer components for the new dash.  We did need to do this for the image answer - it differed significantly and had little to no internal logic outside of the render.  However, for this component, I think it is ok to continue to use the `.js` version instead of adding an extra `.tsx` version.  You'll see that when I was getting the interactive answers to display better inside of our containers, I added a new prop, `responsive` to `iframe-answer.js` to facilitate the very minor changes that I needed to make.  I think you can do the same.  I would continue to use `iframe-answer.js` and add a new prop (which you did already here) to make this one line tweak to the render in this component.  Then we don't have two components that are over 100 lines but that differ by a line or two.